src/core/core/rpc_engine.cpp

```diff
@@ -846,14 +850,28 @@ namespace dsn {
                             {
                                 resolver->on_access_failure(req2->header->gpid.u.partition_index, err);
 
+                                /*
+                                // TODO(qinzuoyan): because the user will fetch reply states from this response_task 'call',
```

Comment:
maybe we can add a rpc_response_task::retry_task() to record the retry
task, and in wait_and unwrap() we check if retry_task_exist.

On Tue, Jul 5, 2016 at 4:13 PM, Zhenyu Guo <notifications@github.com> wrote:

> In src/core/core/rpc_engine.cpp
> <https://github.com/imzhenyu/rDSN/pull/436#discussion_r69522162>:
>
> > @@ -846,14 +850,28 @@ namespace dsn {
> >                              {
> >                                  resolver->on_access_failure(req2->header->gpid.u.partition_index, err);
> >
> > +                                /*
> > +                                // TODO(qinzuoyan): because the user will fetch reply states from this response_task 'call',
>
> You are right, I was thinking about the async case - there are indeed
> issues about the sync case. Let's figure out a solution to it - the retry
> is important otherwise you will always have an error in your first try when
> there is a reconfiguration or it is the code start from the client
> perspective.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/436/files/e9d75c0881fe8a75c3e56362a805f5fe761af0ae#r69522162>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAgQPfKZgeY35NETweedhvzKfTCj0aH1ks5qShI9gaJpZM4JE2Od>
> .
>